src/RevMemCtrl.cc

```diff
@@ -1296,10 +1292,8 @@ void RevBasicMemCtrl::handleReadResp(StandardMem::ReadResp* ev){
         if( isAMO ){
           handleAMO(op);
         }
-        auto Hazard = op->getHazard();
-        if( Hazard != nullptr ){
-          *Hazard = false;
-        }
+        MemReq r = op->getMemReq();
+        r.MarkLoadComplete(r);
```

Comment:
There may be a bug here, but I can't quite tell.  The idea behind this logic is to clear a load hazard in the case where the load spans a cache line (eg, the "split" request).  If we see an AMO occurring, should we clear the hazard?

Reply:
That is a bug - good catch. The handleAMO function will call performAMO() which will clear the hazard. Clearing again here will cause an error. I can add a check similar to line 1316 that only clears if the request is NOT an amo.